test/unit/lib/TopStoriesFeed.test.js

```diff
@@ -1460,4 +1460,38 @@ describe("Top Stories Feed", () => {
     assert.calledOnce(instance.uninit);
     assert.calledOnce(instance.init);
   });
+  describe("#layout", () => {
+    it("should call dispatchLayoutUpdate from loadCachedData and fetchStories", async () => {
+      instance.stories_endpoint = "stories-endpoint";
+      let fetchStub = globals.sandbox.stub();
+
+      const response = {
+        "layout": [1, 2],
+      };
+      globals.set("fetch", fetchStub);
+      fetchStub.resolves({ok: true, status: 200, json: () => Promise.resolve(response)});
+      sinon.spy(instance, "dispatchLayoutUpdate");
+      instance.cache.get = () => ({stories: {layout: [2, 3]}});
+
+      await instance.loadCachedData();
+      assert.calledOnce(instance.dispatchLayoutUpdate);
+      const firstCallArgs = instance.dispatchLayoutUpdate.firstCall.args;
```

Comment:
The firstCall, secondCall argument grovelling makes the test fairly difficult to skim.  If, instead, it were switched to spy.withArgs(), creating a spy that only fired when certain arguments were passed, it would likely be notably more readable.

Reply:
Assuming the above is true, I'd also switch the other newly added tests to this mechanism.

Reply:
Because of splitting it in two, I could easily just run `calledWith` to easily check the values. I couldn't as easily do this before because I was testing a first and second call.

Does that work?

Reply:
Totally.